x-pack/legacy/plugins/fleet/scripts/dev_env_setup/script.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { createFlagError, run } from '@kbn/dev-utils';
+import fetch from 'node-fetch';
+
+interface Policy {
+  id: string;
+}
+
+run(
+  async ({ flags, log }) => {
+    if (flags.kibanaUrl && typeof flags.kibanaUrl !== 'string') {
+      throw createFlagError('please provide a single --kibanaUrl flag');
+    }
+    if (flags.kibanaUser && typeof flags.kibanaUser !== 'string') {
+      throw createFlagError('please provide a single --kibanaUser flag');
+    }
+    if (flags.kibanaPassword && typeof flags.kibanaPassword !== 'string') {
```

Comment:
If you do the assignment to `kibanaPassword` before these checks TS will remember the checks you did and treat `kibanaPassword` as a string automatically. This doesn't extend to properties of objects though, which is why you need to do so many checks below. 

![image](https://user-images.githubusercontent.com/1329312/65463758-3b291500-de0d-11e9-8d55-577c13ffd6b6.png)